CONTRIBUTING.md

```diff
@@ -72,3 +72,74 @@ If you have any doubts about coding standards you can follow [Effective Go](http
 
 Many thanks,
 Kubeshop
+
+
+## Building, running, and e2e testing
+
+As described, the project consists of 3 main components: CLI, server, and web. Each components' directory provides a standard language way to build and test:
+
+- **Web**: uses npm. Commands such as `npm start build`, `npn ci`, etc are available.
+- **CLI**: it is a go module. It can be built with `go build .`, tests are run with `go test ./...` etc
+- **Server**: it is a go module. It can be built with `go build .`, tests are run with `go test ./...` etc
+
+**CLI** and **Server** are mainly built from the project root using [GoReleaser](https://goreleaser.com/) to simplify packaging and distribution.
+
+The server is responsible for serving both the API and the web UI. The easiest way to mix those 2 things for distribution is with docker. We rely on GoReleaser again for packaging all those things together.
```

Comment:
`serving both the CLI and the web UI` ?

Reply:
it doesn't serve the CLI, it serves the API that the CLI consumes. but it's also the "http server" that delivers the web UI (like index.html, *.js, etc). That's what this means by "serve". Not that it is "at the service of", but that it delivers the web responses for.

Does that make sense? Do you think there's a more clear way to state that?

Reply:
I see. No, I think it's ok in that way. Thanks for the details.